pkg/kubefedctl/enable/enable.go

```diff
@@ -181,12 +181,7 @@ func (j *enableType) Run(cmdOut io.Writer, config util.FedConfig) error {
 		return nil
 	}
 
-	if j.DryRun {
-		// Avoid mutating the API
-		return nil
-	}
-
-	return CreateResources(cmdOut, hostConfig, resources, j.KubeFedNamespace)
+	return CreateResources(cmdOut, hostConfig, resources, j.KubeFedNamespace, j.DryRun)
```

Comment:
As a user, I've expected to get feedback durign dry-run about what control plane mutations would happen, so I'm supportive of plumbing through dry-run here.

Reply:
`resources` contains the resources that would be created and the user has the option of providing `--output=yaml` if they want to see that content. I think that should be enough to satisfy said expectations and that the changes proposed here don't provide additional information of value to the user.  

The use of dry run is inconsistent across `kubefedctl` commands, and if the goal is to make that more consistent I'm fine with that.  I think that's an orthogonal concern to fixing the bug indicated in the title of this PR, though, and I think the focus should be on maintaining consistency with the existing way `CreateResources` was being called (i.e. not calling for dry run) rather than what is currently proposed.